optlib/terraform.ctags

```diff
@@ -16,13 +16,20 @@
 # it's way harder to write a good regex for that.
 #   - https://www.terraform.io/docs/language/values/locals.html
 
---langdef=tf
---map-tf=+tf:.tf.tfvars
---regex-tf=/^[[:space:]]*resource[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/r,Resource/
---regex-tf=/^[[:space:]]*data[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/d,Data/
---regex-tf=/^[[:space:]]*variable[[:space:]]*"([^"]*)"/\1/v,Variable/
---regex-tf=/^[[:space:]]*provider[[:space:]]*"([^"]*)"/\1/p,Provider/
---regex-tf=/^[[:space:]]*module[[:space:]]*"([^"]*)"/\1/m,Module/
---regex-tf=/^[[:space:]]*output[[:space:]]*"([^"]*)"/\1/o,Output/
---regex-tf=/^([a-z0-9_]+)[[:space:]]*=/\1/f,TFVar/
+--langdef=terraform
```

Comment:
Sorry, I forgot writing one thing. A parser that is part of ctags should have Capitalized name.
Please, do `s/terraform/Terraform/g`.